src/drf_pydantic/parse.py

```diff
@@ -143,4 +148,38 @@ def _convert_type(type_: type) -> type[serializers.Field]:
     try:
         return FIELD_MAP[type_]
     except KeyError as error:
-        raise NotImplementedError(f"{type_.__name__} is not yet supported") from error
+        try:
+            return FIELD_MAP[type_.__base__]
+        except KeyError:
+            raise NotImplementedError(f"{type_.__name__} is not yet supported") from error
```

Comment:
If you want to chain try-except-KeyError on a dict you can do it as:

```python
field_class = FIELD_MAP.get(type_, {}).get(type_.__base__)
if field_class is None:
    raise NotImplementedError(f"{type_.__name__} is not yet supported")
return field_class
```

This way you don't need to build nested try-excepts.

Also I am not sure if we should be checking for `type_.__base__` - wouldn't the pydantic classes you added earlier be enough?

Reply:
You are right, that was a bad suggestion from me - that would be useful for nested dict traversal, useless here. I have another suggestion:

```python
for key in [type_, type_.__base__]:
    try:
        return FIELD_MAP[key]
    except KeyError:
        continue
raise NotImplementedError(f"{type_.__name__} is not yet supported")
```

Regarding your second point about `__base__`, I think I understand now. Did you mean that `drf_pydantic.base_model.ConstrainedIntValue` has a base of `pydantic.ConstrainedInt`?

Reply:
```python
for key in [type_, type_.__base__]:
    try:
        return FIELD_MAP[key]
    except KeyError:
        continue
raise NotImplementedError(f"{type_.__name__} is not yet supported")
```
Well, that works!

> Did you mean that drf_pydantic.base_model.ConstrainedIntValue has a base of pydantic.ConstrainedInt?

Yes, exactly.